fix: add details to ListUsers for user results

stebenz opened the pull request:
# Which Problems Are Solved

In User v2 API, the ListUsers endpoint doesn't provide the information to which organization the user belongs to.

# How the Problems Are Solved

Add the details to the user results from the ListUsers endpoint, so that the OrgID is also included as ResourceOwner.

# Additional Changes
 
None

# Additional Context

Closes #8172

livio-a commented on internal/api/grpc/user/v2/query_integration_test.go:
```diff
@@ -920,7 +921,10 @@ func TestServer_ListUsers(t *testing.T) {
 					}
+					tt.want.Result[i].Details = infos[i].Details
 				}
+				fmt.Println(tt.want.Result)
+				fmt.Println(got.Result)
```
```suggestion
```

stebenz replied:
removed

livio-a reviewed (changes requested):
how to we deal with the fact, that the userById now returns the details twice:
<img width="788" alt="image" src="https://github.com/zitadel/zitadel/assets/9405495/8718a953-f3ae-496c-9f0d-da50fcccb763">

stebenz commented:
@livio-a saw that as well, was the reason that the test failed, I corrected it and removed it from the GetUserByIDResponse, so that the returned User message is consistent.

stebenz commented:
as mentioned here https://github.com/zitadel/zitadel/issues/8172#issuecomment-2211098049 if the orgID is provided and the passwordChangeRequired already is available in the user v2 calls, this should also be solved
